src/content/blog/if-i-wrote-a-linter-part-3-ecosystem/index.mdx

```diff
@@ -0,0 +1,384 @@
+
+...huh!?
+
+I work on linters and I have a hard time keeping this all straight.
+Now imagine how confusing this all is for someone new to linting, and/or who doesn't care much about their linter.
+
+If I wrote a linter, I would continue the [ESLint Glossary](https://eslint.org/docs/latest/use/core-concepts/glossary) work by defining single recommended terms for all the shared linting concepts.
+I hope this would help reduce naming conflicts such as what we're seeing with _"config"_, _"format"_, and _"stylistic"_.
+
+<LabeledVideo
+	alt="Troy from Community looking confused and upset with his eyes wide open"
+	description="Troy Barnes from Community, attempting to understand these terms."
+	original="https://giphy.com/gifs/community-shocked-surprised-8d72Pydfl4CAw"
+	src={troySittingConfusedZoomIn}
+/>
+
+## Common Core Rules
+
+One of the hardest parts of figuring out a project's ecosystem is knowing what to put in core or delegate to users.
+I've seen two prevailing strategies in the wild:
+
+- Lean core (ESLint): only including rules that pertain to JavaScript itself, which reduces its maintenance burden and allows plugins to iterate and learn
+- Heavy core (Biome, deno lint, Oxlint): including common plugins, which increases the power of the linter out of the box and reduces configuration complexity
+
+I like the lean core approach for allowing the ecosystem to iterate and learn in areas that aren't solidified yet or are too project-dependent for core.
+I don't believe a linter's core should have rules that don't apply to a supermajority of its users.
+Putting a rule in core is a heavyweight action for that rule.
+Development on the rule slows down as the core team has to be more careful about breaking changes than the typical plugin.
+
+On the other hand, once a plugin is known to be stable and applicable to a supermajority of users, putting it in core can be very useful.
+Core linter rules tend to be more stable than plugin rules and more discoverable by virtue of being in the core documentation website.
+Onboarding a plugin to core means users don't have to go out of their way to set it up.
+
+If I wrote a linter, its core rules would pull in all those from plugins that are:
+
+- Common to roughly all projects that would use the linter
+- Stable enough that they are not still rapidly iterating and making breaking changes
+
+That includes taking rules from the following plugins:
+
+- [`@eslint-community/eslint-plugin-eslint-comments`](https://www.npmjs.com/package/@eslint-community/eslint-plugin-eslint-comments)
+- [`eslint-plugin-import`](https://github.com/import-js/eslint-plugin-import)
+- [`eslint-plugin-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc)
+- [`eslint-plugin-package-json`](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json)
+- [`eslint-plugin-promise`](https://github.com/eslint-community/eslint-plugin-promise)
+- [`eslint-plugin-n`](https://github.com/gajus/eslint-plugin-jsdoc)
```

Comment:
I think there's a typo here? Right below it says `n` wouldn't be included in the core.

(I'm enjoying reading this, it sounds like a dream! elm-review but with language plugins :P)